docusaurus/docs/packages/pkg/sdk/sdk.md

```diff
@@ -0,0 +1,345 @@
+
+```go
+type POKTRollSDK interface {
+	GetSessionSupplierEndpoints(
+		ctx context.Context,
+		appAddress string,
+		serviceId string,
+	) (session *SessionSuppliers, err error)
+
+	SendRelay(
+		ctx context.Context,
+		sessionSupplierEndpoint *SingleSupplierEndpoint,
+		request *http.Request,
+	) (response *servicetypes.RelayResponse, err error)
+}
+```
+
+The interface, its implementation and the relevant types involved could be found
+at the following directory [pkg/sdk/](https://github.com/pokt-network/poktroll/blob/main/pkg/sdk)
+
+```
+pkg/sdk/
+├── deps_builder.go   // Logic for auto-building the dependencies of the SDK
+├── errors.go         // Errors returned by the SDK
+├── interface.go      // POKTRollSDK interface
+├── relay_verifier.go // Logic to verify relay responses
+├── sdk.go            // POKTRollSDK implementation and initialization function
+├── send_relay.go     // Logic to send relay requests
+├── session.go        // Logic to handle sessions and retrieve the suppliers list
+└── urls.go           // Helpers to parse the urls used by the SDK to interact with the Pocket Network
+```
+
+### GetSessionSupplierEndpoints
+
+`GetSessionSupplierEndpoints` returns a `SessionSuppliers` struct that contains
+the fully-hydrated session corresponding to the `appAddress` and `serviceId`
+provided and a list of `SingleSupplierEndpoint`s, where each `SingleSupplierEndpoint`
+contains the necessary information to send a relay request to the `Supplier` it represents.
+
+The `POKTRollSDK` consumer is free to run any strategy to select a `Supplier` from
+the list returned by the `GetSessionSupplierEndpoints` method.
+
+### SendRelay
+
+Once a `Supplier` is selected, the `POKTRollSDK` consumer can then pass the corresponding
+`SingleSupplierEndpoint` to the `SendRelay` method which will take care of constructing
+the `RelayRequest`, sending it to the `Supplier` and verifying the response signature.
+
+Once the `RelayRequest` is sent, the `SendRelay` method will wait for the `RelayResponse`
+and verify its signature before returning it to the `POKTRollSDK` consumer.
+
+:::warning
+
+The error returned by the `SendRelay` only indicates that an error occurred during
+the process of sending the `RelayRequest` or verifying the `RelayResponse` signature
+and does not indicate errors returned by the `Supplier` or the requested service.
+These ones should be available in the `RelayResponse` returned by the `SendRelay`
+and the `POKTRollSDK` considers them as valid responses.
+
+:::
+
+### NewPOKTRollSDK
+
+`NewPOKTRollSDK` is the initializing function that returns a fully functional
+`POKTRollSDK` implementation given it's provided a valid `POKTRollSDKConfig` struct
+which contains the necessary information to build the dependencies of the
+`POKTRollSDK` implementation.
+
+```go
+func NewPOKTRollSDK(
+  ctx context.Context,
+  config *POKTRollSDKConfig,
+) (POKTRollSDK, error)
+```
+
+### POKTRollSDKConfig
```

Comment:
@red-0ne Do you think we should have a TODO to enable setting this via a yaml file as well?

Reply:
@Olshansk , I don't think the `SDK` should set its config "directly" via config files.

Since the SDK is primarily intended to be consumed by other packages, it's up to them to source the config params from files. Just like we're doing with the `AppGateServer` which is actually sourcing from config files then passing these to the SDK.